File: frontend/lib/features/auth/domain/user_role.dart
enum UserRole {
  platformOwner('platform_owner'),
  institutionAdmin('institution_admin'),
  teacher('teacher'),
  student('student'),
  parent('parent');

  const UserRole(this.value);

  final String value;

  static UserRole parse(String value) {
    for (final role in values) {
      if (role.value == value) {
        return role;
      }
    }

    throw FormatException('Unknown user role: $value');
  }
}

File: backend/app/Enums/UserRole.php
<?php

namespace App\Enums;

enum UserRole: string
{
    case PlatformOwner = 'platform_owner';
    case InstitutionAdmin = 'institution_admin';
    case Teacher = 'teacher';
    case Student = 'student';
    case Parent = 'parent';

    /**
     * @return list<string>
     */
    public static function values(): array
    {
        return array_column(self::cases(), 'value');
    }
}
